CMakeLists.txt

```diff
@@ -37,10 +38,30 @@ string(REPLACE ";" ", " IRODS_DEVELOP_DEPENDENCIES_STRING "${IRODS_DEVELOP_DEPEN
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(IRODS_CXX_STANDARD "${CMAKE_CXX_STANDARD}")
+
 # export-dynamic so stacktrace entries from executables have function names
-set(CMAKE_EXE_LINKER_FLAGS_INIT "-Wl,--export-dynamic -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,defs")
-set(CMAKE_MODULE_LINKER_FLAGS_INIT "-Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,defs")
-set(CMAKE_SHARED_LINKER_FLAGS_INIT "-Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,defs")
+if (IRODS_ENABLE_ADDRESS_SANITIZER)
+  add_compile_options(
+    -fsanitize=address
+    -fno-omit-frame-pointer
+    -fno-optimize-sibling-calls
+    -O1
+  )
+  add_link_options(
+    -fsanitize=address
+    -fno-omit-frame-pointer
+    -fno-optimize-sibling-calls
+    -O1
+  )
```

Comment:
Consider if additional options are needed.
Consider whether -O0 would be better.

Reply:
Marking as resolved as these options work just fine.

We can always tweak later.